src/network_analysis.jl

```diff
@@ -886,3 +886,45 @@ function treeweight(t::SimpleDiGraph, g::SimpleDiGraph, distmx::Matrix)
     end
     prod
 end
+
+"""
+    cycles(rs::ReactionSystem)
+
+    Returns the matrix of cycles (or flux vectors), or reaction fluxes at steady state. These correspond to right eigenvectors of the stoichiometric matrix. Equivalent to [`fluxmodebasis`](@ref). 
+"""
+
+function cycles(rs::ReactionSystem)
+    nps = get_networkproperties(rs)
+    nsm = netstoichmat(rs)
+    !isempty(nps.cyclemat) && return nps.cyclemat
+    nps.cyclemat = cycles(nsm; col_order = nps.col_order)
+    nps.cyclemat
+end
+
+function cycles(nsm::T; col_order = nothing) where {T <: AbstractMatrix}
+
+    # compute the left nullspace over the integers
+    N = MT.nullspace(nsm; col_order)
+
+    # if all coefficients for a cycle are negative, make positive
+    for Nrow in eachcol(N)
+        all(r -> r <= 0, Nrow) && (Nrow .*= -1)
+    end
+
+    # check we haven't overflowed
+    iszero(nsm * N) || error("Calculation of the cycle matrix was inaccurate, "
+          * "likely due to numerical overflow. Please use a larger integer "
+          * "type like Int128 or BigInt for the net stoichiometry matrix.")
+
+    T(N)
+end
```

Comment:
I think it would make sense to factor out the code that is common to conservation laws and cycles, and just handle passing it the transposed matrix for conservation laws. i.e. make a common function both calculations wrap given they are essentially identical up to the transposes.